model-optimizer/mo/middle/passes/fusing/fuse_linear_ops.py

```diff
@@ -111,6 +111,17 @@ def _fuse_mul(graph: Graph, node: Node, fuse_nodes: list, backward: bool = True)
         weights_port.get_connection().set_source(w_mul.out_port(0))
         w_const.connect(w_mul.in_port(tensor_port.idx))
 
+        fuse_node_in_data = fuse_node.in_node(weights_port.idx)
+        w_const_out_data = w_const.node.out_node(w_const.idx)
+
+        # During this reconnection new data node name is copied from the data node
+        # outgoing from w_const port. Duplicate names of data nodes lead to appear
+        # of duplicate op node names after constant folding. So we should manually
+        # set a unique name for the new data node.
+        if fuse_node_in_data.soft_get('name') == w_const_out_data.soft_get('name') and \
+                fuse_node_in_data.soft_get('name', None) is not None:
+            fuse_node.in_node(weights_port.idx)['name'] = graph.unique_id()
```

Comment:
Can we add `fuse_node_in_data.soft_get('name')` or `w_const_out_data.soft_get('name')` as a parameter to the `unique_id` method so the name of the generated node is more meaningful?

Reply:
fuse_node_in_data.soft_get('name') and w_const_out_data.soft_get('name') contain op name of w_const node + Output_0/Data_. unique_id() will add some postfix, but when this name will be used for Const op creation, the postfix will be removed with Output_0/Data_ and the resulting name will be the same as original op name. Which leads to "Not unique names" error. 
So I suggest using name of w_mul, as w_mul produces this data node and doesn't contain name identical to other op nodes.